src/main/java/com/healthyfood/connectivity/FetchApiFood.java

```diff
@@ -0,0 +1,60 @@
+package com.healthyfood.connectivity;
+
+import com.healthyfood.utils.CapitalizeTheFirstLetter;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Collections;
+import java.util.List;
+
+public class FetchApiFood {
+    private final HttpClient client = HttpClient.newBuilder().build();
+
+    public String getProductByName(String productName) {
+        try {
+            HttpRequest request = HttpRequest.newBuilder().uri(URI.create("https://api.edamam.com/api/food-database/v2/parser?app_id=a5c1a6e2&app_key=%20917d9904d1aa8302eb4d353787034eb6&ingr=" + CapitalizeTheFirstLetter.capitalizeTheFirstLetter(productName) + "&nutrition-type=cooking")).GET().build();
+            return getResponds(request);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public String getProductByFoodId(String foodId) {
+        try {
+            HttpRequest request = HttpRequest.newBuilder().uri(URI.create("https://api.edamam.com/api/food-database/v2/parser?app_id=a5c1a6e2&app_key=%20917d9904d1aa8302eb4d353787034eb6&ingr=" + foodId + "&nutrition-type=cooking")).GET().build();
+            return getResponds(request);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public List<String> getAllProducts() {
+        try {
+            HttpRequest request = HttpRequest.newBuilder().uri(URI.create("https://api.edamam.com/api/food-database/v2/parser?app_id=a5c1a6e2&app_key=%20917d9904d1aa8302eb4d353787034eb6")).GET().build();
+            return Collections.singletonList(getResponds(request));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
Just  a note: if you're returning a list or any type of collections, never return null, rather an empty list. In this case it would be best to return Collections.emptyList()

Reply:
Thank you for the great tip.